### PR TITLE
#95: Improve CI rendering tests

### LINKED ISSUE #95: Improve CI rendering tests
Follows up of #32 

- [x] Improve output testing (VTP, PNG)
- [x] Fix invalid badge in README

### PR DESCRIPTION
**Pull request by tlamonthezie**
Fixes #95

CI improvements in continuation to issue #32 PR #85 in particular for rendering tests (VTP and PNG).

This PR provides
- Better image and mesh testing: set `object_jitter` value to `0.0` for test configurations to disable randomness in point placement. Then the rendering tests can be efficient to test output files against expected (\*.vtp|\*.png)
- Improve testing code quality and output
- Move datasets from `{project_dir}/tests/data` datasets to `{project_dir}/data`
- Fix tests badge url in README.md
- Add XVBF usage as an option of build.sh for running tests

### CODE REVIEW

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -5,6 +5,7 @@ on:
   push:
     branches:
       - master
+      - 95-improve-ci-rendering-tests # TODO: REMOVE ONCE PR CREATED
```
Comment (cwschilly) [inline .github/workflows/build-and-test.yml:8]: Just a reminder to remove this when appropriate

File: build.sh [hunk 2]
```diff
@@ -40,6 +40,7 @@ VT_TV_COVERAGE_ENABLED=$(on_off ${VT_TV_COVERAGE_ENABLED:-OFF})
 VT_TV_CLEAN=$(on_off ${VT_TV_CLEAN:-ON})
 VT_TV_PYTHON_BINDINGS_ENABLED=$(on_off ${VT_TV_PYTHON_BINDINGS_ENABLED:-OFF})
 VT_TV_WERROR_ENABLED=$(on_off ${VT_TV_WERROR_ENABLED:-OFF})
+VT_TV_XVFB_ENABLED=$(on_off ${VT_TV_WERROR_ENABLED:-OFF})
```
Comment (cwschilly) [inline build.sh:43]:
```suggestion
VT_TV_XVFB_ENABLED=$(on_off ${VT_TV_XVFB_ENABLED:-OFF})
```

File: ci/build.sh
Comment (cwschilly) [inline ci/build.sh:1]: Can we delete this file?

File: src/vt-tv/render/render.h [hunk 3]
```diff
@@ -97,6 +97,7 @@
 #include <array>
 #include <variant>
 #include <cmath>
+#include <regex>
```
Comment (cwschilly) [inline src/vt-tv/render/render.h:100]: Why is this include needed?

File: tests/unit/render/test_parse_render.cc [hunk 4]
```diff
@@ -62,70 +68,47 @@ class ParseRenderTest :public ::testing::TestWithParam<std::string> {
 
   void SetUp() override {
     // Make the output directory for these tests
-    std::filesystem::create_directory(fmt::format("{}/output", SRC_DIR));
-    std::filesystem::create_directory(fmt::format("{}/output/tests", SRC_DIR));
+    // std::filesystem::create_directory(fmt::format("{}/output", SRC_DIR));
+    // std::filesystem::create_directory(fmt::format("{}/output/tests", SRC_DIR));
+    // std::filesystem::create_directory(fmt::format("{}/output/tests/no_png", SRC_DIR));
   }
 };
```
Comment (cwschilly) [inline tests/unit/render/test_parse_render.cc:75]: Can we remove these lines?

File: tests/unit/render/test_render.cc [hunk 5]
```diff
@@ -65,44 +71,157 @@ class RenderTest :public ::testing::TestWithParam<std::string> {
 
   void SetUp() override {
     // Make the output directory for these tests
-    std::filesystem::create_directory(fmt::format("{}/output", SRC_DIR));
-    std::filesystem::create_directory(fmt::format("{}/output/tests", SRC_DIR));
-    std::filesystem::create_directory(fmt::format("{}/output/tests/render", SRC_DIR));
+    // std::filesystem::create_directory(fmt::format("{}/output", SRC_DIR));
+    // std::filesystem::create_directory(fmt::format("{}/output/tests", SRC_DIR));
+  }
```
Comment (cwschilly) [inline tests/unit/render/test_render.cc:76]: Same here--is the `SetUp` necessary still?

File: ci/build.sh
Comment (tlamonthezie) [reply ci/build.sh:1]: In fact no. This file is calling the root build script but set arguments values required by CI

File: tests/unit/render/test_render.cc (hunk 5 above)
Comment (tlamonthezie) [reply tests/unit/render/test_render.cc:76]: method removed

File: .github/workflows/build-and-test.yml (hunk 1 above)
Comment (tlamonthezie) [reply .github/workflows/build-and-test.yml:8]: you are right this has been removed

File: build.sh (hunk 2 above)
Comment (tlamonthezie) [reply build.sh:43]: Update done

File: src/vt-tv/render/render.h (hunk 3 above)
Comment (tlamonthezie) [reply src/vt-tv/render/render.h]: this has been removed

File: tests/unit/deps/test_vtk.cc [hunk 6]
```diff
@@ -81,49 +81,49 @@ class VtkTest :public ::testing::Test {
...
+  w2i->ReadFrontBufferOff();       // read from the back buffer
+  w2i->Update();
 
   vtkNew<vtkPNGWriter> writer;
   writer->SetFileName(fmt::format("{}/output/tests/vtk_example_screenshot.png", SRC_DIR).c_str());
-  writer->SetInputConnection(windowToImageFilter->GetOutputPort());
+  writer->SetInputConnection(w2i->GetOutputPort());
 }
 
 } // end namespace vt::tv::tests::unit
```
Comment (cwschilly) [inline tests/unit/deps/test_vtk.cc:129]: Missing newline

**Review by cwschilly — CHANGES_REQUESTED** [review]

File: tests/unit/deps/test_vtk.cc (hunk 6 above)
Comment (tlamonthezie) [reply tests/unit/deps/test_vtk.cc]: Done

**Review by cwschilly — APPROVED** [review]
LGTM

**Pull request merged**